templates/Civi/Angular/Page/Main.tpl

```diff
@@ -1,5 +1,11 @@
+{if $route}
+  <script type="text/javascript">
+    location.hash = '{$route}';
```

Comment:
@totten, I passed this from PHP to JS using \CRM_Core_Resources::singleton()->addSetting instead. Before this change, /civicrm/a/?route=%27;%20alert("sucker");%20%27 would execute JS on the page. After this change, it does not.